modules/renter/contractor/persist.go

```diff
@@ -120,6 +133,26 @@ func (c *Contractor) load() error {
 		}
 	}
 
+	// COMPATv1.1.0
+	//
+	// Wwhen loading old contracts, the corresponding relationships may not be
+	// known. Use the hostdb to make sure the relationship set is fully filled
+	// out.
+	allHosts := c.hdb.AllHosts()
+	hmap := make(map[modules.NetAddress]modules.HostDBEntry)
+	// Iterate backwards (lowest score to highest) so that in the event of
+	// duplicate hosts for a netaddress, the highest score host is the one in
+	// the map.
+	for i := len(allHosts) - 1; i >= 0; i-- {
+		hmap[allHosts[i].NetAddress] = allHosts[i]
+	}
```

Comment:
interesting, but maybe too clever, since it relies on allHosts being properly sorted. You can do this explicitly at the cost of verbosity:
```go
for _, h := range c.hdb.AllHosts() {
    // in the event of a duplicate address, use the host with the highest score
    if dup, exists := hmap[h.NetAddress]; !exists || dup.Score < h.Score {
        hmap[h.NetAddress] = h
    }
}
```
Although for that to work, you'd need an external way of determining the score...